splitio/storage/pluggable.py

```diff
@@ -207,3 +207,170 @@ def put(self, split):
         if existing_split is not None and existing_split.traffic_type_name != split.traffic_type_name:
             self._increase_traffic_type_count(split.traffic_type_name)
             self._decrease_traffic_type_count(existing_split.traffic_type_name)
+
+
+class PluggableSegmentStorage(SegmentStorage):
+    """Pluggable implementation of segment storage."""
+    _SEGMENT_NAME_LEMNGTH = 14
+    _TILL_LENGTH = 4
+
+    def __init__(self, pluggable_adapter, prefix=None):
+        """Constructor."""
+        self._pluggable_adapter = pluggable_adapter
+        self._prefix = "SPLITIO.segment.{segment_name}"
+        self._segment_till_prefix = "SPLITIO.segment.{segment_name}.till"
+        if prefix is not None:
+            self._prefix = prefix + "." + self._prefix
+            self._segment_till_prefix = prefix + "." + self._segment_till_prefix
+
+    def update(self, segment_name, to_add, to_remove, change_number=None):
+        """
+        Update a segment. Create it if it doesn't exist.
+
+        :param segment_name: Name of the segment to update.
+        :type segment_name: str
+        :param to_add: Set of members to add to the segment.
+        :type to_add: set
+        :param to_remove: List of members to remove from the segment.
+        :type to_remove: Set
+        """
+        try:
+            self._pluggable_adapter.add_items(self._prefix.format(segment_name=segment_name), to_add)
+            self._pluggable_adapter.remove_items(self._prefix.format(segment_name=segment_name), to_remove)
+            if change_number is not None:
+                self._pluggable_adapter.set(self._segment_till_prefix.format(segment_name=segment_name), change_number)
+        except Exception:
+            _LOGGER.error('Error updating segment storage')
+            _LOGGER.debug('Error: ', exc_info=True)
+
+    def set_change_number(self, segment_name, change_number):
```

Comment:
You can replace line 241 by calling `set_change_number`. Actually, I think you can also remove `set_change_number` method, if the change number is only update in line 241.

Reply:
I will reuse set_change_number, it is in the spec, so we need to keep it.